Fix color picker bug

unlocomqx opened the pull request:
When multiple pickers are in the page, they all will have the same background color even if they don't have the same color value

<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | When multiple color pickers are in the page, they will all have the same background color even if they have different color values
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #25013
| How to test?      | You will need a page with multiple pickers, I used my module page as an exmple but couldn't find PrestaShop pages having multiple pickers


<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/25012)
<!-- Reviewable:end -->

unlocomqx committed:
Fix color picker bug

When multiple pickers are in the page, they all will have the same background color even if they don't have the same color value

kpodemski commented:
I'm not sure if fixing it directly in third-party library is a way to go, maybe binding color picker should be done in the loop for every `mColorPicker` element?

unlocomqx commented:
@kpodemski I tried that, but it had the same effect, because the color is updated for `$('.mColorPicker')` so it will take whatever color is in the closure and apply it to all `.mColorPicker` inputs. 
I fixed the third party lib because an edit on this lib caused the bug, it wasn't present in previous versions.
Btw, I manages to reproduce it on the "Order Notifications on the Favicon" module (a PrestaShop official module)

Before

<img width="765" alt="2021-06-18 at 8 20 AM" src="https://user-images.githubusercontent.com/793712/122523782-4f646b00-d00f-11eb-8f90-941e0e17c051.png">

After

<img width="752" alt="2021-06-18 at 8 25 AM" src="https://user-images.githubusercontent.com/793712/122523802-55f2e280-d00f-11eb-9610-5d136f3b71e0.png">

atomiix commented on js/jquery/plugins/jquery.colorpicker.js:
```diff
@@ -43,7 +43,7 @@
     
     $(document).ready(function() {
-      $.fn.mColorPicker.setTextColor($('.mColorPicker'));
+     $.fn.mColorPicker.setTextColor($(this));
```
```suggestion
      $.fn.mColorPicker.setTextColor($(this));
```

unlocomqx replied:
Reindented code

atomiix commented:
@unlocomqx Thanks for your contribution! It would be great if you could provide a sample module to try it out so the QA team can validate your PR easily. Do you think you could do it?

NeOMakinG commented:
> @unlocomqx Thanks for your contribution! It would be great if you could provide a sample module to try it out so the QA team can validate your PR easily. Do you think you could do it?

Hey @atomiix , @unlocomqx said that we could reproduce it with Order Notifications on the Favicon (a PS module)

unlocomqx commented:
> @unlocomqx Thanks for your contribution! It would be great if you could provide a sample module to try it out so the QA team can validate your PR easily. Do you think you could do it?

I included screenshots from a native module above

Thanks

hibatallahAouadni commented:
Hello @unlocomqx 

Thanks for your PR! It works:

![image](https://user-images.githubusercontent.com/23308427/123084966-06068800-d422-11eb-86a1-3fa763cc580d.png)

But, I have a question, why in the multistore page **create a shop**, the color picker was white:

![image](https://user-images.githubusercontent.com/23308427/123085181-3ea66180-d422-11eb-9813-97cb5db21ee4.png)

With your PR, it turns black:

![image](https://user-images.githubusercontent.com/23308427/123085238-4cf47d80-d422-11eb-8e3f-56edfc009938.png)

Why? @NeOMakinG any idea?
Please check and feedback.

Thanks!

unlocomqx committed:
Update the newly created inputs on document ready

This will solve the black background color when the input is empty

unlocomqx commented:
@hibatallahAouadni I added a new commit to fix the issue

The reason was that the [code](https://github.com/PrestaShop/PrestaShop/pull/25012/commits/2ad154383045612360a408761af1944d12fcad3d#diff-2e3746771eaa16666e95c99394d3f07083d124a59889ddb861394c99198debbfR46) was updating the original color input, but that input has been removed at that point and replaced with the text input.
The solution is to keep track of the new inputs then update them correctly

hibatallahAouadni commented:
Hello @unlocomqx 

LGTM, QA :heavy_check_mark: 
Thanks!

Progi1984 commented:
Thanks @unlocomqx &  @hibatallahAouadni

PrestaEdit commented:
Hi all !

For informations, this issue was introduced inside the 1.7.8 PrestaShop version, it was working before.
Why does it reach the 8.0 version to be fixed, as this ?

PrestaEdit commented:
And this is the Pull Request that introduce this: https://github.com/PrestaShop/PrestaShop/pull/20148

Progi1984 commented:
@PrestaShop/qa-functional Hi (or Good Night), could you confirm what @PrestaEdit says ? If it is the case, it should be a regression. Isn't it ?

hibatallahAouadni commented:
Hello everyone,

The issue related used to be Bug and I don't know why it changed to Feature :sweat: 
That's why it was fixed on 8.0.0 and not on 1.7.8.0.
Well, I need to verify it if it's a regression on 1.7.7.8, cause with PS1767 and PS1773 I didn't reproduce the issue.

Thanks for your patience :pray:

hibatallahAouadni commented:
@kpodemski @PrestaEdit @Progi1984 indeed it's a regression on PS1.7.8.0 and it should be fixed on 1.7.8.x.
Can we cherry-picked this PR to fix the issue on 1.7.8.x?
